trl/trainer/stepwise_reward_trainer.py

```diff
@@ -0,0 +1,297 @@
+            )
+        elif is_peft_available() and peft_config is not None:
+            if not isinstance(model, PeftModel):
+                if getattr(model, "is_loaded_in_8bit", False) or getattr(model, "is_quantized", False):
+                    _supports_gc_kwargs = "gradient_checkpointing_kwargs" in list(
+                        inspect.signature(prepare_model_for_kbit_training).parameters
+                    )
+
+                    prepare_model_kwargs = {"use_gradient_checkpointing": args.gradient_checkpointing}
+
+                    if not _supports_gc_kwargs and args.gradient_checkpointing_kwargs is not None:
+                        warnings.warn(
+                            "You passed `gradient_checkpointing_kwargs` in the trainer's kwargs, but your peft version does not support it. "
+                            "please update to the latest version of peft to use `gradient_checkpointing_kwargs`."
+                        )
+                    elif _supports_gc_kwargs and args.gradient_checkpointing_kwargs is not None:
+                        prepare_model_kwargs["gradient_checkpointing_kwargs"] = args.gradient_checkpointing_kwargs
+
+                    model = prepare_model_for_kbit_training(model, **prepare_model_kwargs)
+
+                model = get_peft_model(model, peft_config)
+
+        if compute_metrics is None:
+            compute_metrics = compute_accuracy
+
+        if data_collator is None:
+            if tokenizer is None:
+                raise ValueError(
+                    "A tokenizer must be specified when using the default DataCollatorForTokenClassification"
+                )
+            data_collator = DataCollatorForTokenClassification(tokenizer, max_length=args.max_length)
+
+        if "input_ids" not in train_dataset.column_names:
+            if args.max_length is None:
+                args.max_length = 512
+                warnings.warn(
+                    "When the dataset isn't pretokenized, you should set `max_length` in the `StepwiseRewardConfig`"
+                    " we have set it for you, but you should do it yourself in the future."
+                )
+            if args.step_separator is None:
+                args.step_separator = "\n"
+                warnings.warn(
+                    "When the dataset isn't pretokenized, you should set `step_separator` in the `StepwiseRewardConfig`"
+                    " we have set it for you to '\n', but you should do it yourself in the future."
+                )
+            with PartialState().local_main_process_first():
+                chat_template_kwargs = {"tokenizer": tokenizer}
+                tokenize_kwargs = {
+                    "tokenizer": tokenizer,
+                    "max_length": args.max_length,
+                    "step_separator": args.step_separator,
+                }
+                train_dataset = train_dataset.map(maybe_apply_chat_template, fn_kwargs=chat_template_kwargs)
```

Comment:
add `num_proc=args.dataset_num_proc,` for this one as well. (same for eval dataset)

I realise that it's missing in `RewardTrainer` as well, I'll fix it